ckan/logic/action/get.py

```diff
@@ -2070,15 +2075,29 @@ def resource_search(context, data_dict):
             split_terms[field] = terms
         fields = split_terms
 
+    include_private = asbool(data_dict.get('include_private', False))
+
+    # return a version of `q` which limits access to private datasets,
+    # appropriate to the access level of the current user
+    def _protect_private(q):
+        if include_private and authz.is_sysadmin(user):
+            return q
+        elif include_private and user:
```

Comment:
You could drop the `elif` and the and `user`.

You're returning early above above and less conditions -> easier to understand.